Get 1.14 to sync initial blocks

rnicoll opened the pull request:
This is a very skeletal patch to just get 1.14 to the point where it launches and can sync some blocks on testnet. AuxPoW definitely doesn't work, Digishield probably doesn't work, and there's a LOT of values that need fixing still, this is just enough to get the ball rolling.

rnicoll commented:
@patricklodder I've ended up squashing your test correction commit into one I've made, and it's tagged me as author for both. Let me know if you'd rather take blame for commit 091634e

DarrenTa commented:
Couldn't the change of amount.h lead to  a chain fork in the event that a transaction over 21,000,000 is made?

rnicoll commented:
@DarrenTa No, because what we're doing is updating Bitcoin to match Dogecoin parameters, rather than updating an existing Dogecoin codebase to match Bitcoin changes. The amount change brings this in line with the Dogecoin Core 1.11 value: https://github.com/dogecoin/dogecoin/blob/master/src/amount.h#L20

DarrenTa commented:
@rnicoll Thanks, your citation makes it clear.  Thank you again.

patricklodder commented:
@rnicoll No need to put extra effort to correct the blame unless you prefer to blame :-)

langerhans commented on src/chainparams.cpp:
```diff
@@ -107,32 +107,33 @@ class CMainParams : public CChainParams {
-        base58Prefixes[EXT_SECRET_KEY] = boost::assign::list_of(0x04)(0x88)(0xAD)(0xE4).convert_to_container<std::vector<unsigned char> >();
-
-        vFixedSeeds = std::vector<SeedSpec6>(pnSeed6_main, pnSeed6_main + ARRAYLEN(pnSeed6_main));
+        vSeeds.push_back(CDNSSeedData("dogecoin.com", "seed.dogecoin.com", true));
```
This one is offline

langerhans commented on src/chainparams.cpp:
```diff
@@ -107,32 +107,33 @@ class CMainParams : public CChainParams {
+        vSeeds.push_back(CDNSSeedData("dogecoin.com", "seed.dogecoin.com", true));
+        vSeeds.push_back(CDNSSeedData("multidoge.org", "seed.multidoge.org", true));
+        vSeeds.push_back(CDNSSeedData("multidoge.org", "seed2.multidoge.org"));
+        vSeeds.push_back(CDNSSeedData("doger.dogecoin.com", "seed.doger.dogecoin.com", true));
```
This one too. We should think about adding at least one new one.

langerhans commented on doc/build-osx.md:
```diff
@@ -16,7 +16,8 @@ Then install [Homebrew](http://brew.sh).
 
-    brew install automake berkeley-db4 libtool boost --c++11 miniupnpc openssl pkg-config protobuf --c++11 qt5 libevent
+    brew install automake libtool boost --c++11 miniupnpc openssl pkg-config protobuf --c++11 qt5 libevent
+    brew install berkeley-db # You need to make sure you install a version >= 5.1.29, but as close to 5.1.29 as possible. Check the homebrew docs to find out how to install older versions.
```
Do we want to pull docs changes so early? I think it would be beneficial to keep most of the cosmetic details separate as we did before. Makes picking changes later easier I guess? But I'm open for anything ;)

rnicoll commented:
I'm pulling changes from 1.10 manually, but squashing a lot of them down as I go. I've reverted the docs for now, removed the dead seeds, and annotated a bunch of chain params we need to investigate later.

rnicoll commented:
Also fixed the wallet tests that depend on fixed rewards, to match the Dogecoin reward schedule. If anyone wants to tackle the last failing test that would be great, but otherwise eager to just get this merged.
